### PR TITLE
ANN: Highlight code that is unsafe

### PR DESCRIPTION
**Pull request by garyttierney**
Adds a border around expressions that are unsafe within an `unsafe
{}` block or unsafe function, to create a distinction between safe and unsafe
code.

The current default color settings show a red border around the reference name to the unsafe code, like so:

screenshot from 2018-11-24 00-28-16

Though we might want to consider something else (it was originally the same as Kotlin smart casts, cc @vlad20012).

First pass attempt at #3013.

### CODE REVIEW

**Comment by vlad20012** [comment]
Some tests are failed

**Comment by garyttierney** [comment]
I missed the annotation for unsafe method calls, will update.  Maybe I should add tests for the info annotation too (and possibly feature flag it behind a toggle?)

**Comment by Undin** [comment]
@garyttierney do you need any help with it?

**Comment by garyttierney** [comment]
@Undin no, thank you though! I'll have tests up over the weekend, it's just been a lack of time on my part :)

**Comment by garyttierney** [comment]
Hm, applying `FFE8C6` as a foreground color doesn't result in the best visibility (and also doesn't seem to work outside of the preview, maybe because the syntax highlighter annotator also applies a foreground?):
screenshot from 2018-12-17 21-52-14

Though I think this is similar to the color commonly used for warnings, so I've attempted to apply it as the background. That looks better:

screenshot from 2018-12-17 21-51-29

**Comment by garyttierney** [comment]
It's difficult to write tests for info annotations since that's also what the highlighter uses, so almost every token has an annotation associated with it that needs to be in the test fixture. Should we maybe make these annotations weak warnings instead?

**Comment by vlad20012** [comment]
Test failed

**Comment by garyttierney** [comment]
Should be OK now.

File: src/main/resources/org/rust/ide/colors/RustDefault.xml [hunk 1]
```diff
@@ -38,4 +38,9 @@
             <option name="FOREGROUND" value="CBCBCB" />
         </value>
     </option>
+    <option name="org.rust.UNSAFE_CODE">
+        <value>
+            <option name="EFFECT_COLOR" value="ff0030" />
```
[the remark this answers is not in the document]
Comment (Undin) [reply src/main/resources/org/rust/ide/colors/RustDefault.xml]: @garyttierney Oh, of course, I meant background. Sorry, for misleading

**Comment by Undin** [comment]
> Should we maybe make these annotations weak warnings instead?

**Comment by Undin** [comment]
Thanks a lot!
bors r+

**Comment by garyttierney** [comment]
@Undin no problem. I think if the highlighting annotator can be explicitly disabled under test it'd be easier to write tests for other info annotations, but I couldn't find a way to do that.

Thanks for the review!

**Pull request merged**